posthog/api/hog_function.py

```diff
@@ -160,7 +160,8 @@ def validate(self, attrs):
         template = HogFunctionTemplates.template(template_id) if template_id else None
 
         if hog_type == "transformation":
-            if not settings.HOG_TRANSFORMATIONS_CUSTOM_HOG_ENABLED:
+            # Only allow team ID 2 to edit transformation code, even when the flag is enabled
+            if not settings.HOG_TRANSFORMATIONS_CUSTOM_HOG_ENABLED or team.id != 2:
```

Comment:
This isn't the way to do this. Swap it to a setting called `HOG_TRANSFORMATIONS_CUSTOM_ENABLED_TEAMS: "2,3,etc"` 

That way we can control it via an env var as it should be